kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/core/v1/PodOperationsImpl.java

```diff
@@ -315,10 +315,10 @@ private URL getURL(String operation, String[] commands) throws MalformedURLExcep
    */
   String validateOrDefaultContainerId(String name) {
     Pod pod = this.require();
-    List<Container> containers = pod.getSpec().getContainers();
-    if (containers.isEmpty()) {
+    if (pod.getSpec() == null || pod.getSpec().getContainers() == null || pod.getSpec().getContainers().isEmpty()) {
       throw new KubernetesClientException("Pod has no containers!");
```

Comment:
I'm not sure that this is possible, I added this after implementing a test that failed (using the mockserver).
If it is possible to create a Pod without spec or containers, then we should definitely handle this

Reply:
No, I don't believe it was possible.  But I wasn't sure how pervasive you were trying to cover those cases.

Reply:
So my thoughts:
- I don't think this resource can exist in a real cluster (Pod | PodTemplate) -I did a very quick test-
- I added this here because a test in the MockServer this resource can exist.
- I don't think it hurts having this guards here, although the most correct approach would be to create a valid resource in the test and remove these.
- I think it's fine leaving this here, but I can remove if you think it's inconsistent or that this might be confusing.

Please let me know what do you prefer

Reply:
How about a comment explaining that it's for making testing easier.  There's less danger it will be unnecessarily copied later that way.

Reply:
sounds good